flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/join/stream/AbstractStreamingJoinOperator.java

```diff
@@ -117,7 +117,9 @@ public void open() throws Exception {
 	@Override
 	public void close() throws Exception {
 		super.close();
-		joinCondition.backingJoinCondition.close();
+		if (joinCondition != null) {
```

Comment:
Any change to not init `joinCondition`?

Reply:
I think we should protect it in each close() method. The compiled instance may not be instantiated when close() is invoked (an exception happens before instantiation). 

This change is not related to this pull request.